modules/graphman-operation-validate.js

```diff
@@ -1,24 +1,72 @@
 const utils = require("./graphman-utils");
-const graphman = require("./graphman");
+const policySchema = require("./policy-schema");
+
 var error = false;
+var policyErrors = false;
 module.exports = {
+    schemaValidator: null,
     run: function (params) {
         if (params.input) {
+            buildPolicySchema()
             const obj = utils.readFile(params.input)
-            if (Object.keys(obj).length == 1 && Object.keys(obj)[0] == "All") {
-                validateAssertions(Object.values(obj)[0])
-                if (!error) {
-                    utils.info("validation is successful");
-                }
-            } else {
-                console.log("Given policy is invalid");
+            validateBundle(obj);
+            if (!error) {
+                utils.info("validation is successful");
             }
         }
     },
 
     usage: function () {
-        console.log("    validate --input <input-file>");
+        utils.print("    validate --input <input-file>");
+    }
+}
+
+function validateBundle(obj) {
+    for (const [entityName, entities] of Object.entries(obj)) {
+        if (entityName == "services" || entityName == "policies") {
+            validateEntities(entities)
+        }
     }
+
+}
+
+function validateEntities(entities) {
+    for (const entity of entities) {
+        policyErrors = false
+        var name
+        for (const [propertyName, propertyValue] of Object.entries(entity)) {
+            if (propertyName == "name") {
+                name = propertyValue
+            } else if (propertyName == "policy") {
+                validatePolicy(propertyValue)
+            }
+        }
+        if (policyErrors) {
+            utils.info("Invalid policy code in " + name);
+        }
+    }
+}
+
+function validatePolicy(policy) {
+    for (const [propertyName, propertyValue] of Object.entries(policy)) {
```

Comment:
looping can be avoided if we go by policy.code or policy.json

Reply:
done